qtpy/QtWidgets.py

```diff
@@ -8,15 +8,27 @@
 
 """Provides widget classes and functions."""
 
-from . import PYQT5, PYQT6, PYSIDE2, PYSIDE6
+from . import PYQT5, PYQT6, PYSIDE2, PYSIDE6, QtModuleNotInstalledError
+
+
+MISSING_SYMBOL_ERRORS = {}
+
 
 if PYQT5:
     from PyQt5.QtWidgets import *
 elif PYQT6:
     from PyQt6 import QtWidgets
     from PyQt6.QtWidgets import *
     from PyQt6.QtGui import QAction, QActionGroup, QShortcut, QFileSystemModel, QUndoCommand
-    from PyQt6.QtOpenGLWidgets import QOpenGLWidget
+
+    # Attempt to import QOpenGLWidget, but if that fails, don't throw an exception until the
+    # symbol is explicitly asked for
+    # this allows removing the sizeable OpenGL binaries from pyinstaller bundles, but
+    # maintains compatibility with apps that expect this symbol to be automatically imported
+    try:
+        from PyQt6.QtOpenGLWidgets import QOpenGLWidget
+    except (ModuleNotFoundError, ImportError):
+        MISSING_SYMBOL_ERRORS['QOpenGLWidget'] = QtModuleNotInstalledError(name='PyQt6.QtOpenGLWidgets', missing_package='pyopengl')
```

Comment:
This alternate style puts the error site in _complete_ control of what gets raised, instead of just the message contents.

I think I prefer this because all the 'thinking' is up here, and the `__getattr__` is now purely mechanical.

It's possible that the (potential) reuse of an exception instance is dangerous somehow; I couldn't find an answer to this online. Making this a lambda should be a sufficient workaround, if necessary.